File: CSharp/MainMenu.cs
using System;

namespace ConsoleMenu
{
    public class MainMenu : Menu
    {
        public MainMenu()
        {
            HeaderText = "Main Menu";

            Add("Print \"Hellow World\"", PrintHelloWorld);
            Add("Print this item name", ThisItemName);
        }

        private delegate void WrapHandler();

        private void Wrapper(WrapHandler handler)
        {
            Console.Clear();
            Console.CursorVisible = true;

            handler();

            Console.WriteLine();
            Console.WriteLine("Press any key to continue!");
            Console.ReadKey();
        }

        private void PrintHelloWorld(MenuItem sender)
        {
            Wrapper(() =>
            {
                Console.WriteLine("Hello world");
            });
        }

        private void ThisItemName(MenuItem sender)
        {
            Wrapper(() =>
            {
                Console.WriteLine(sender.Name);
            });
        }
    }
}

File: CPlusPlus/main_menu.cpp
#include <iomanip>
#include <sstream>
#include "main_menu.h"

MainMenu::MainMenu()
{
    setSize(74, 24);
    setHeaderText("Main Menu");

    ADD_ITEM("Print \"Hello World\"", &MainMenu::printHelloWorld);
    ADD_ITEM("Print this item name", &MainMenu::printThisItemName);
}

int MainMenu::wrapper(function<int()> func)
{
    Console::clear();
    Console::showCursor();

    int result = func();

    if (!result)
    {
        cout << endl;
        cout << "Press any key to continue!" << endl;
        Console::readKey();
    }

    return result;
}

int MainMenu::printHelloWorld(MenuItem* sender)
{
    return wrapper([]()
        {
            cout << "Hello World" << endl;
            return 0;
        });
}

int MainMenu::printThisItemName(MenuItem* sender)
{
    return wrapper([]()
        {
            cout << "Print this item name" << endl;
            return 0;
        });
}